### ISSUE TITLE
Regarding some hyperparameters for training

### ISSUE THREAD

**Issue by zjysteven** [P2333/Adaptive-Diversity-Promoting#2]
Hi, thanks for making the code public. I'm trying to reproduce your experiments with Pytorch. However, I find that there is some mismatch between the hyperparameters specified in your code and those mentioned in the paper. For example, in the paper, the training of CIFAR-10 took 180 epochs, while in the code it will take 200 epochs and the learning rate schedule is also set accordingly (with milestones at 100, 150). I just want to have a confirmation from you on this. Thanks!

**Comment by zjysteven** [comment]
Also, I'm wondering what is the purpose of this line when calculating the loss.
https://github.com/P2333/Adaptive-Diversity-Promoting/blob/10d77e79bdf494358a9c776ea10d3e9cb9d58ae3/utils.py#L95
Taking `num_model=3` as an example, `y_true` should originally be a batch of concatenated one-hot labels with shape [batch_size, 3*10], and `tf.reduce_sum(y_true, axis=1)` should be 3, right? If this is the case, I can't see why this line is necessary since it seems to change nothing.

**Comment by P2333** [comment]
Hi, thank you for your interest. The results reported in the paper on CIFAR-10 train for 180 epochs with learning rate decay at 120 and 160 epochs. We adjust the setting in code to 200 (100,150) epochs because this is more standard and consistent with other works. Besides, 200 (100,150) would have slightly better performance than the original 180 (120,160) setting.

About the first line in the loss function, it seems you are right and it is embarrassing that I cannot remember the reason for writing it here. So maybe you can ignore this line, at least when all the labels are one-hot vectors.

**Comment by zjysteven** [comment]
Thanks for the clarification! It helps!

**Comment by zjysteven** [comment]
Sorry to bother you again, but I just want to confirm another thing. When measuring the white-box robustness of the ensemble, it seems that you are using the model predictions instead of the ground-truth labels to generate adversarial attacks, right?

Specifically, only the inputs are passed into the attack function as shown below, and then by default cleverhans would use model predictions as the label.
https://github.com/P2333/Adaptive-Diversity-Promoting/blob/10d77e79bdf494358a9c776ea10d3e9cb9d58ae3/test_cifar_iterative.py#L143

**Comment by P2333** [comment]
Yes, the model predictions are used in attacks to avoid label leakage. The evaluation is fair as long as the adversarial attacking processes are consistent for baseline defenses and the proposed defense.

Of course, you can use the labels to craft adversarial examples more rigorously, but this would not cause a significant difference since the clean accuracy is usually high on CIFAR-10 or MNIST. Things could be different on, e.g., ImageNet when the predicted clean labels probably differ from the true ones.
